packages/nextjs/app/builders/0xe9Ad7D1C2069E0Fa9b5852Adc77C9196651BB8b8/page.tsx

```diff
@@ -0,0 +1,63 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import Image from "next/image";
+import Socials from "./components/Socials";
+import type { NextPage } from "next";
+import { useEnsAvatar, useEnsName } from "wagmi";
+import { Address } from "~~/components/scaffold-eth";
+
+const builderAddress = "0xe9Ad7D1C2069E0Fa9b5852Adc77C9196651BB8b8";
+
+const KrvvsProfile: NextPage = () => {
+  const [ensAvatar, setEnsAvatar] = useState<string | null>();
+
+  const { data: fetchedEns } = useEnsName({
+    address: builderAddress,
+    chainId: 1,
+  });
+
+  const { data: fetchedEnsAvatar } = useEnsAvatar({
+    name: fetchedEns ? fetchedEns : "",
+    chainId: 1,
+  });
+
+  useEffect(() => {
+    setEnsAvatar(fetchedEnsAvatar);
+  }, [fetchedEnsAvatar]);
```

Comment:
I think, we can remove useState and useEffect hooks for ensAvatar. We can directly use the data from useEnsAvatar hook, because it already manages its own state internally.